api/base/parsers.py

```diff
@@ -15,6 +17,29 @@ class JSONAPIParser(JSONParser):
     media_type = 'application/vnd.api+json'
     renderer_class = JSONAPIRenderer
 
+    def flatten_relationships(self, relationships):
+        """
+        Flattens relationships dictionary which has information needed to create related resource objects.
+
+        Validates that formatting of relationships dictionary is correct.
+        """
+        if not isinstance(relationships, dict):
+                raise ParseError()
```

Comment:
Glad this is fixed.

As a side note, if the seeming disappearance of a function wasn't caught by unit tests, it may point to a gap in test coverage. (not sure how thoroughly we test formatting / JSONAPI compliance)